posit-dev/positron
# Add UI modal dialog for saving plots

### Intent
Address #2169 

### Approach
For static plots, present a file dialog to select the save location. It saves the data uri that is shown in the plots pane.

For dynamic plots, a dialog is presented to select the save location, width, height, and DPI. A preview is available if the user wants to see what it looks like before saving. At save, the plot is rendered and saved.

The modal dialog uses callbacks as was changed with the recent refactoring with modals.

The file formats was mostly scaffolding for future support and this continues that with a bit of hardcoding to PNG.

![image](https://github.com/posit-dev/positron/assets/9591545/28e64c3d-566a-45f9-a904-46d22cbefa91)


### QA Notes
See https://docs.google.com/document/d/1CqA8Ypji8FhiqZ5MujvilfV58btoNfIbsjK67qFtvd8/edit?usp=sharing for the spec.

The dialog does validate the values before generating a preview and saving. This wasn't tried on Windows so there may be bugs with the path validation that may need fixing.

The file format is PNG only at the moment
